core/chains/evm/txm/orchestrator.go

```diff
@@ -0,0 +1,358 @@
+	return &Orchestrator[BLOCK_HASH, HEAD]{
+		lggr:           logger.Sugared(logger.Named(lggr, "Orchestrator")),
+		chainID:        chainID,
+		txm:            txm,
+		txStore:        txStore,
+		keystore:       keystore,
+		attemptBuilder: attemptBuilder,
+		fwdMgr:         fwdMgr,
+	}
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) Start(ctx context.Context) error {
+	return o.StartOnce("Orchestrator", func() error {
+		var ms services.MultiStart
+		if err := ms.Start(ctx, o.attemptBuilder); err != nil {
+			return fmt.Errorf("Orchestrator: AttemptBuilder failed to start: %w", err)
+		}
+		addresses, err := o.keystore.EnabledAddressesForChain(ctx, o.chainID)
+		if err != nil {
+			return err
+		}
+		for _, address := range addresses {
+			err := o.txStore.Add(address)
+			if err != nil {
+				return err
+			}
+		}
+		if err := ms.Start(ctx, o.txm); err != nil {
+			return fmt.Errorf("Orchestrator: Txm failed to start: %w", err)
+		}
+		if o.fwdMgr != nil {
+			if err := ms.Start(ctx, o.fwdMgr); err != nil {
+				return fmt.Errorf("Orchestrator: ForwarderManager failed to start: %w", err)
+			}
+		}
+		return nil
+	})
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) Close() (merr error) {
+	return o.StopOnce("Orchestrator", func() error {
+		if o.fwdMgr != nil {
+			if err := o.fwdMgr.Close(); err != nil {
+				merr = errors.Join(merr, fmt.Errorf("Orchestrator failed to stop ForwarderManager: %w", err))
+			}
+		}
+		if err := o.attemptBuilder.Close(); err != nil {
+			merr = errors.Join(merr, fmt.Errorf("Orchestrator failed to stop AttemptBuilder: %w", err))
+		}
+		if err := o.txm.Close(); err != nil {
+			merr = errors.Join(merr, fmt.Errorf("Orchestrator failed to stop Txm: %w", err))
+		}
+		return merr
+	})
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) Trigger(addr common.Address) {
+	o.txm.Trigger(addr)
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) Name() string {
+	return o.lggr.Name()
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) HealthReport() map[string]error {
+	return map[string]error{o.Name(): o.Healthy()}
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) RegisterResumeCallback(fn txmgr.ResumeCallback) {
+	o.resumeCallback = fn
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) Reset(addr common.Address, abandon bool) error {
+	ok := o.IfStarted(func() {
+		if err := o.txm.Abandon(addr); err != nil {
+			o.lggr.Error(err)
+		}
+	})
+	if !ok {
+		return errors.New("Orchestrator not started yet")
+	}
+	return nil
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) OnNewLongestChain(ctx context.Context, head HEAD) {
+	ok := o.IfStarted(func() {
+		o.attemptBuilder.OnNewLongestChain(ctx, head)
+	})
+	if !ok {
+		o.lggr.Debugw("Not started; ignoring head", "head", head, "state", o.State())
+	}
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) CreateTransaction(ctx context.Context, request txmgrtypes.TxRequest[common.Address, common.Hash]) (tx txmgrtypes.Tx[*big.Int, common.Address, common.Hash, common.Hash, evmtypes.Nonce, gas.EvmFee], err error) {
```

Comment:
Why is this function not exactly similar to the CreateTransaction() in txmgr.go?
That had some more things like:
- checking for checkEnabled().
- txStore.CheckTxQueueCapacity()
- txRequest.Strategy.PruneQueue

Also, I didn't fully understand why we need a wrapped Tx, and extra logic for pipelineTaskRunID and meta json marshaling

Reply:
Unfortunately, we can't do that because the two methods use two different underlying tx structures. TXM has generics whereas TXMv2 doesn't. We would have to make changes to the existing TXM to do that, which is something we don't want to do. The difference in the pruning mechanism comes from the fact that as of now, the InMemory layer TXMv2 uses already has an embedded pruning mechanism.